### PR TITLE
Add dedicated Redis for GOV.UK chat in integration/staging/prod

### PR DESCRIPTION
**Pull request by rtrinque**
To reduce the blast radius of GOV.UK chat we use a separated Redis cluster for it.

### CODE REVIEW

File: terraform/deployments/chat/redis.tf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  subnet_ids = local.elasticache_subnets
+}
+
+resource "aws_security_group" "chat_redis_cluster" {
+  name        = local.chat_redis_name
+  vpc_id      = data.tfe_outputs.vpc.nonsensitive_values.id
+  description = "GOV.UK Chat Redis cluster"
+  tags = {
+    Name   = local.chat_redis_name
+    System = "Sidekiq"
```
Comment (kevindew) [inline terraform/deployments/chat/redis.tf:17]: We are expecting that we'll use Redis for other things than Sidekiq, such as tracking rate limits.

File: terraform/deployments/tfc-configuration/variables-integration.tf [hunk 2]
```diff
@@ -55,6 +55,22 @@ module "variable-set-integration" {
...
 
+module "variable-set-chat-integration" {
+  source = "./variable-set"
+
+  name = "chat-integration"
+
+  tfvars = {
+    chat_redis_cluster_apply_immediately          = true
+    chat_redis_cluster_automatic_failover_enabled = false
+    chat_redis_cluster_engine_version             = "6.x"
```
Comment (kevindew) [inline terraform/deployments/tfc-configuration/variables-integration.tf:66]: Shall we use version 7? That's what we use in dev and on Heroku.

**Review by kevindew — COMMENTED** [review]
Couple of minor comments.

File: terraform/deployments/tfc-configuration/variables-integration.tf (hunk 2 above)
Comment (rtrinque) [reply terraform/deployments/tfc-configuration/variables-integration.tf]: Changed engine version to 7.1

File: terraform/deployments/chat/redis.tf (hunk 1 above)
Comment (rtrinque) [reply terraform/deployments/chat/redis.tf:17]: Right, I guess we can remove this tag for now, I'm not sure it's used for anything beyond being descriptive.

Comment (rtrinque) [reply terraform/deployments/chat/redis.tf]: removed

File: terraform/deployments/chat/redis.tf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  description = "GOV.UK Chat Redis cluster"
+  tags = {
+    Name = local.chat_redis_name
+  }
+}
+
+resource "aws_elasticache_replication_group" "chat_redis_cluster" {
+  apply_immediately          = var.chat_redis_cluster_apply_immediately
+  replication_group_id       = local.chat_redis_name
+  description                = "Redis for Sidekiq queues"
```
Comment (kevindew) [inline terraform/deployments/chat/redis.tf:23]: Just another sidekiq reference to sort out
```suggestion
  description                = "Redis for GOV.UK Chat"
```

**Pull request merged**